ansible_collections/arista/avd/plugins/action/yaml_templates_to_facts.py

```diff
@@ -135,26 +134,18 @@ def run(self, tmp=None, task_vars=None):
             elif "python_module" in template_item:
                 module_path = template_item.get("python_module")
                 class_name = template_item.get("python_class_name", DEFAULT_PYTHON_CLASS_NAME)
+                cls = load_python_class(module_path, class_name, AvdFacts)
 
-                try:
-                    cls = getattr(importlib.import_module(module_path), class_name)
-                except ImportError as imp_exc:
-                    raise AnsibleActionFail(imp_exc) from imp_exc
-
-                if issubclass(cls, AvdFacts):
-                    cls_instance = cls(hostvars=template_vars, templar=templar)
-                    if debug:
-                        debug_item["timestamps"]["render_python_class"] = datetime.now()
-                    if getattr(cls_instance, "render"):
-                        try:
-                            template_result_data = cls_instance.render()
-                        except Exception as error:
-                            raise AnsibleActionFail(error) from error
-                    else:
-                        raise AristaAvdError(f"{cls_instance} has no attribute render")
+                cls_instance = cls(hostvars=template_vars, templar=templar)
+                if debug:
+                    debug_item["timestamps"]["render_python_class"] = datetime.now()
+                if getattr(cls_instance, "render"):
```

Comment:
this check should not be reversed to raise inside if (avoid indentation of the main action)